File: day19/src/main/java/de/havox_design/aoc2017/day19/Direction.java
package de.havox_design.aoc2017.day19;

public enum Direction {
    UP,
    RIGHT,
    DOWN,
    LEFT;

    public Direction left() {
        return switch (this) {
            case LEFT -> DOWN;
            case DOWN -> RIGHT;
            case RIGHT -> UP;
            case UP -> LEFT;
        };
    }

    public Direction right() {
        return switch (this) {
            case LEFT -> UP;
            case DOWN -> LEFT;
            case RIGHT -> DOWN;
            case UP -> RIGHT;
        };
    }
}

File: day21/src/main/kotlin/de/havox_design/aoc2023/day21/Direction.kt
package de.havox_design.aoc2023.day21

enum class Direction(val direction: Pair<Int, Int>) {
    DOWN(Pair(0, 1)),
    LEFT(Pair(-1, 0)),
    RIGHT(Pair(1, 0)),
    UP(Pair(0, -1))
}